### ISSUE TITLE
wrapper with matPrefix or matSuffix, does not change internal values at runtime.

### ISSUE THREAD

**Issue by GlauberF** [ngx-formly/ngx-formly#2961]
**Description**
today I have a wrapper to show a loading or icon, where would be matPrefix or matSuffix.
Where I show based on the value passed inside templateOptions.

example for loading.
```
this.field.templateOptions = {
      ...this.field.templateOptions,
      loading: {
          show: true
      }
  };
```
My wrapper

HTML
```
<!-- Component Type -->
<ng-container #fieldComponent></ng-container>

<!-- MAT SUFFIX -->
<ng-template #matSuffix>
    <ng-container *ngIf="to.matSuffix || to.loading?.show">

        <!-- Loading -->
        <ng-container *ngIf="to.loading?.show">
            <div class="vimbo-mat-addons">
                <span class="spacer"></span>
                <span class="vimbo-mat-addons-suffix loading" [ngClass]="to.loading?.className" [ngStyle]="to.loading?.theme">
                    <mat-progress-spinner
                        class="mt-2"
                        [color]="'accent'"
                        [mode]="'indeterminate'"
                        [diameter]="20">
                    </mat-progress-spinner>
                </span>
            </div>
        </ng-container>

        <!-- Icon, text ... -->
        <ng-container *ngIf="to.matSuffix && !to.loading?.show">
            <div class="vimbo-mat-addons" [ngStyle]="{cursor: to.matSuffix.onClick ? 'pointer' : 'inherit'}" (click)="matSuffixClick($event)">
                <span class="spacer"></span>
                <span class="vimbo-mat-addons-suffix" [ngClass]="{'no-bg-color': to.matSuffix?.noBackground}" [ngStyle]="to.matSuffix?.theme">
                    <ng-container *ngIf="to?.matSuffix?.text">
                        <span [ngClass]="to?.matSuffix?.optionsText?.className" [ngStyle]="to?.matSuffix?.optionsText?.theme">
                            {{to.matSuffix?.text}}
                        </span>
                    </ng-container>
                    <ng-container *ngIf="to?.matSuffix?.matIcon">
                        <mat-icon [ngClass]="to?.matSuffix?.optionsIcons?.className" [ngStyle]="to?.matSuffix?.optionsIcons?.theme">
                            {{to?.matSuffix?.matIcon}}
                        </mat-icon>
                    </ng-container>
                    <ng-container *ngIf="to?.matSuffix?.matIconSvg">
                       <mat-icon svgIcon="{{to?.matSuffix?.matIconSvg}}" [ngClass]="to?.matSuffix?.optionsIcons?.className" [ngStyle]="to?.matSuffix?.optionsIcons?.theme">
                       </mat-icon>
                    </ng-container>
                </span>
            </div>
        </ng-container>

    </ng-container>
</ng-template>
<!-- // MAT SUFFIX -->
```
Typescript
```
export class VimboFormlyWrapperAddonsComponent extends FieldWrapper implements OnInit, AfterViewInit, OnDestroy {
    @ViewChild('matPrefix', { static: true }) matPrefix: TemplateRef<any>;
    @ViewChild('matSuffix', { static: true }) matSuffix: TemplateRef<any>;

ngAfterViewInit(): void {
        // Is Server
        if (typeof window === 'undefined') {
            return;
        }

        // if (this.matPrefix) {
        //     Promise.resolve().then(() => (this.to.prefix = this.matPrefix));
        // }
        //
        // if (this.matSuffix) {
        //     Promise.resolve().then(() => (this.to.suffix = this.matSuffix));
        // }
        if (this.matPrefix) {
            this.to.prefix = this.matPrefix;
        }

        if (this.matSuffix) {
            this.to.suffix = this.matSuffix;
        }
    }
}
```

And this is important, because we will always change something, some templateOption value and that must be reflected at runtime.

As in the case of loading, which can be linked to a variable and this variable can change between true/false, according to what the user types in a field of type autocomplete

**Minimal Reproduction**
my scenario is more complex than playback, but you can see that after setTimeout runs the new value for addonRight doesn't reflect on the screen
https://stackblitz.com/edit/angular-wczz8l?file=src%2Fapp%2Fapp.component.ts

**Your Environment**
```
Angular CLI: 11.2.12
Node: 14.16.1
OS: linux x64

Formly: 6.0.0-next.1

Angular: 11.2.13
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... platform-server, router, service-worker
Ivy Workspace: Yes

Package                            Version
------------------------------------------------------------
@angular-devkit/architect          0.1102.11
@angular-devkit/build-angular      0.1102.12
@angular-devkit/build-optimizer    0.1101.4
@angular-devkit/core               11.2.11
@angular-devkit/schematics         11.2.12
@angular/cdk                       11.2.12
@angular/cli                       11.2.12
@angular/flex-layout               11.0.0-beta.33
@angular/material                  11.2.12
@angular/material-moment-adapter   11.2.12
@nguniversal/builders              11.2.1
@nguniversal/express-engine        11.2.1
@schematics/angular                11.2.12
@schematics/update                 0.1102.12
rxjs                               6.6.7
typescript                         4.1.5
```

**Comment by aitboudad** [comment]
This issue has been fixed and released as part of [v6.0.0-next.4](https://github.com/ngx-formly/ngx-formly/releases/tag/v6.0.0-next.4) release.

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!

**Comment by nauseouscore-ui** [comment]
> This issue has been fixed and released as part of [v6.0.0-next.4](https://github.com/ngx-formly/ngx-formly/releases/tag/v6.0.0-next.4) release.
> 
> Please let us know, in case you are still encountering a similar issue/problem. Thank you!





https://stylesheets.space/questions/adjust-the-color-of-the-icon-depending-on-the-value
